yjit/src/core.rs

```diff
@@ -1875,11 +1879,16 @@ fn get_branch_target(
 
         // Add an incoming branch into this block
         block.push_incoming(branchref.clone());
-        let mut branch = branchref.borrow_mut();
-        branch.blocks[target_idx.as_usize()] = Some(blockref.clone());
```

Comment:
If blocks[idx] is set, then dst_addrs[idx] should also be set. In the old code, the caller is responsible for doing it, which made it harder to see that these two are closely related.